api/common/ApiRequestContext.ts

```diff
@@ -66,3 +78,26 @@ export const authorized: AzureFunction = async function (
 
   await wrappedFunction(ctx);
 };
+
+async function hasPermission(username: string, context: Context, permission: string): Promise<boolean> {
```

Comment:
While this is fine - it's going to incur an extra round trip to the db every time we issue an authorised API request. This should probably happen when a user logs in and has their jwt token issued - their roles should be set in the token, and we should just validate their presence here.

Reply:
Sure that makes sense, if we merge in these 3 PRs I can then create a new PR that reworks the location checked for the role and the addition of the role to the generated token.